lib/simulator-xcode-8.js

```diff
@@ -29,6 +30,49 @@ class SimulatorXcode8 extends SimulatorXcode7 {
   }
 
   /**
+   * @return {string} Bundle identifier of Simulator UI client.
+   */
+  get uiClientBundleId () {
+    return UI_CLIENT_BUNDLE_ID;
+  }
+
+  /**
+   * Check the state of Simulator UI client.
+   * @Override
+   *
+   * @return {boolean} True of if UI client is running or false otherwise.
+   */
+  async isUIClientRunning () {
+    const args = ['-e', `tell application "System Events" to count processes whose bundle identifier is "${this.uiClientBundleId}"`];
+    const {stdout} = await exec('osascript', args);
```

Comment:
Does this need any special permissions for the terminal (or whatever is running the Appium process)?